src/backend/joanie/core/models/courses.py

```diff
@@ -279,19 +282,37 @@ def delete(self, using=None):
         also triggers on query deletes and is being called as many times as there are objects in
         the query. This would generate many separate calls to the webhook and would not scale. We
         decided to not provide synchronization for the moment on bulk deletes and leave it up to
-        the developper to handle these cases correctly.
+        the developer to handle these cases correctly.
         """
+        # course run will be deleted, we synchronize it setting the course run hidden
+        self.synchronize_course_run(self, hidden=True)
         super().delete(using)
-        self.synchronize_with_webhooks()
+        # we now synchronize the course run created for the products
+        self.synchronize_with_webhooks(list_hook=[enums.HOOK_MODEL_PRODUCT])
 
-    def synchronize_with_webhooks(self):
+    def synchronize_with_webhooks(self, list_hook=None):
```

Comment:
I feel this pattern of passing "list_hook" is wrong but I don't get why it was necessary?